test/ServerSubscriptionTest.php

```diff
@@ -48,30 +52,30 @@ function tearDown()
     function testBasic() {
 
         $subscriptionData = new Subscription();
-        $subscriptionData->accountId =  $this->account->accountId;
-        $subscriptionData->productName = "Sports";
-        $subscriptionData->productCategory = "BASE";
-        $subscriptionData->billingPeriod = "MONTHLY";
-        $subscriptionData->priceList = "DEFAULT";
-        $subscriptionData->externalKey = $this->externalBundleId;
+        $subscriptionData->setAccountId($this->account->getAccountId());
+        $subscriptionData->setProductName('Sports');
+        $subscriptionData->setProductCategory('BASE');
+        $subscriptionData->setBillingPeriod('MONTHLY');
+        $subscriptionData->setPriceList('DEFAULT');
+        $subscriptionData->setExternalKey($this->externalBundleId);
 
         $subscription = $subscriptionData->create($this->user, $this->reason, $this->comment, $this->tenant->getTenantHeaders());
-        $this->assertEquals($subscription->accountId, $subscriptionData->accountId);
-        $this->assertEquals($subscription->productName, $subscriptionData->productName);
-        $this->assertEquals($subscription->productCategory, $subscriptionData->productCategory);
-        $this->assertEquals($subscription->billingPeriod, $subscriptionData->billingPeriod);
-        $this->assertEquals($subscription->externalKey, $subscriptionData->externalKey);
+        $this->assertEquals($subscription->getAccountId(), $subscriptionData->getAccountId());
+        $this->assertEquals($subscription->getProductName(), $subscriptionData->getProductName());
+        $this->assertEquals($subscription->getProductCategory(), $subscriptionData->getProductCategory());
+        $this->assertEquals($subscription->getBillingPeriod(), $subscriptionData->getBillingPeriod());
+        $this->assertEquals($subscription->getExternalKey(), $subscriptionData->getExternalKey());
 
         # Move by a few days -- still in trial -- and change product
         $this->clock->addDays(3, $this->tenant->getTenantHeaders());
-        $subscription->productName = 'Super';
+        $subscription->setProductName('Super');
         $subscriptionRes = $subscription->changePlan(null, null, null, $this->user, $this->reason, $this->comment, $this->tenant->getTenantHeaders());
-        $this->assertEquals($subscriptionRes->productName, 'Super');
+        $this->assertEquals($subscriptionRes->getProductName(), 'Super');
```

Comment:
> Can you reproduce this somehow?

Yes, finally! 😅 

For anyone following at home, here's the magic command to run a single php test with phpunit installed through composer:

```
./vendor/phpunit/phpunit/phpunit --filter testBasic Killbill\Client\ServerSubscriptionTest test/ServerSubscriptionTest.php
```

Bonus points for updating the README for non-PHP devs 😼 

Looking at the server log, it looks like the plan isn't changed to a `Super`:

```
2017-04-06T14:17:50,706+0000 lvl='INFO', log='DefaultEntitlement', th='qtp1492980244-33', xff='', rId='4e6eee99-131f-43b0-b934-427bcc337173', aRId='65858', tRId='6056', Change Entitlement Plan:  id = '7a9740fd-c511-49da-84bc-d0dee836658e', plan='sports-monthly', overrides='[(:sports-monthly-trial:USD:0:null),(:sports-monthly-evergreen:USD:null:500)]'
```

The issue is that the new plan name should be passed (`super-monthly`) instead of changing the product name in the test.